### PR TITLE
✨ Enabled Aggregate API for Virtual Cluster

### PR DESCRIPTION
**Pull request by gyliu513**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubernetes-sigs/cluster-api-provider-nested/issues/166

@christopherhein @Fei-Guo 

This PR only covered VirtualCluster but does not cover nested control plane, will fix nested control plane in another PR.

@wangjsty ^^

### CODE REVIEW

File: controlplane/nested/component-templates/nested-apiserver/nested-apiserver-statefulset-template.yaml [hunk 1]
```diff
@@ -55,6 +55,14 @@ spec:
...
         - --apiserver-count=1
         - --endpoint-reconciler-type=master-count
+        - --enable-aggregator-routing=true
+        - --requestheader-client-ca-file=/etc/kubernetes/pki/root/tls.crt
+        - --requestheader-allowed-names=""
+        - --requestheader-username-headers=X-Remote-User
+        - --requestheader-group-headers=X-Remote-Group
+        - --requestheader-extra-headers-prefix=X-Remote-Extra-
+        - --proxy-client-key-file=/etc/kubernetes/pki/apiserver/tls.key
+        - --proxy-client-cert-file=/etc/kubernetes/pki/apiserver/tls.crt
```
Comment (christopherhein) [inline controlplane/nested/component-templates/nested-apiserver/nested-apiserver-statefulset-template.yaml:65]: It's not suggested to use the same CA and thus Cert/Key as the APIServer check out - https://kubernetes.io/docs/tasks/extend-kubernetes/configure-aggregation-layer/#ca-reusage-and-conflicts for more information, we actually already create a new CA/Cert/Key for "front-proxy" which is meant to be used for this purpose - https://github.com/kubernetes-sigs/cluster-api-provider-nested/blob/main/controlplane/nested/controllers/nestedapiserver_controller.go#L228-L246 you'll need to add this to the volume mounts to add support for it.

**Review by christopherhein — CHANGES_REQUESTED** [review]
Once change, otherwise this looks good, thank you for adding support for these.

File: controlplane/nested/component-templates/nested-apiserver/nested-apiserver-statefulset-template.yaml (hunk 1 above)
Comment (gyliu513) [reply controlplane/nested/component-templates/nested-apiserver/nested-apiserver-statefulset-template.yaml]: Done

**Comment by christopherhein** [comment]
Did you mean to remove the changes you made in `controlplane/nested/component-templates/nested-apiserver/nested-apiserver-statefulset-template.yaml` prior?

**Comment by gyliu513** [comment]
> Did you mean to remove the changes you made in controlplane/nested/component-templates/nested-apiserver/nested-apiserver-statefulset-template.yaml prior?

@christopherhein for now, yes, I want to use this PR for virtual cluster and then create another PR for nested control plane soon, hope it is OK.

**Comment by christopherhein** [comment]
Sounds good, just wanted to make sure. :+1:
/lgtm
/approve

**Review by christopherhein — APPROVED** [review]

**Comment by wangjsty** [comment]
@gyliu513 @christopherhein， One more comment even through the PR was merged.
I validated the PR again, looks like it's good, but please `- --requestheader-allowed-names=front-proxy-client` to replace the current `- --requestheader-allowed-names=""`. Thanks.

**Comment by christopherhein** [comment]
> @gyliu513 @christopherhein， One more comment even through the PR was merged.
> I validated the PR again, looks like it's good, but please use `- --requestheader-allowed-names=front-proxy-client` to replace the current `- --requestheader-allowed-names=""`. Thanks.

Good find, @wangjsty can you make sure an issue is filled and @gyliu513 can you do this as well?

**Comment by gyliu513** [comment]
@christopherhein Done at https://github.com/kubernetes-sigs/cluster-api-provider-nested/pull/179

**Pull request merged**